src/core/lib/iomgr/tcp_posix.cc

```diff
@@ -576,6 +576,8 @@ static bool tcp_flush(grpc_exec_ctx* exec_ctx, grpc_tcp* tcp,
       if (errno == EAGAIN) {
         tcp->outgoing_slice_idx = unwind_slice_idx;
         tcp->outgoing_byte_idx = unwind_byte_idx;
+        grpc_slice_buffer_partial_unref_internal(exec_ctx, tcp->outgoing_buffer,
```

Comment:
If we have not sent any (sent_len < 0), I do not think we need to unref here.

Reply:
My reason for doing the unreffing here is that this is the point in the write loop where we might break out of this function, so right before we do that, I want to unref everything that was written out in this call to tcp_flush. I am not actually unreffing slice[unwind_slice_idx], because that didn't get written (because of sent_len < 0), I am just unreffing everything before it